src/classes/MbugoHelpers.cls

```diff
@@ -54,220 +50,166 @@ public with sharing class MbugoHelpers {
         totalVslaMaleFarmerMetricWrapper.setLabel('Number of male members registered in the farmer learning groups');
-        metricData = getMetricData(TOTAL_NUMBER_OF_MALE_FARMERS_IN_VSLAS);
-        totalVslaMaleFarmerMetricWrapper.setMetricId(String.valueOf(metricData.Id));
-        totalVslaMaleFarmerMetricWrapper.setCurrentTarget(metricData.Projected_Value__c);
-        totalVslaMaleFarmerMetricWrapper.setCurrentRealValue((Decimal) metricData.Manual_Value__c);
-
-
-
-        MetricDataWrapper totalVslaFemaleFarmerMetricWrapper = new MetricDataWrapper();
-        totalVslaFemaleFarmerMetricWrapper.setLabel('Number of female members registered in the farmer learning groups');
-        metricData = getMetricData(TOTAL_NUMBER_OF_FEMALE_FARMERS_IN_VSLAS);
-        totalVslaFemaleFarmerMetricWrapper.setMetricId(String.valueOf(metricData.Id));
-        totalVslaFemaleFarmerMetricWrapper.setCurrentTarget(metricData.Projected_Value__c);
-        totalVslaFemaleFarmerMetricWrapper.setCurrentRealValue((Decimal) metricData.Manual_Value__c);
-
-
-
-
-        List<MetricDataWrapper> wrapper = new List<MetricDataWrapper> {vslaMetricWrapper, totalVslaFarmerMetricWrapper, 
-            totalVslaMaleFarmerMetricWrapper, totalVslaFemaleFarmerMetricWrapper};
-    
-        return wrapper;
+        try {
+        	// TODO: Get these from db  
+        	metricData = getMetricData(NUMBER_OF_VSLAS_REGISTERED);
+            vslaMetricWrapper.setMetricId(String.valueOf(metricData.Id));
+            vslaMetricWrapper.setCurrentTarget(metricData.Projected_Value__c);
+            vslaMetricWrapper.setCurrentRealValue((Decimal) metricData.Manual_Value__c);
+            // TODO: Get these from db  
+            metricData = getMetricData(TOTAL_NUMBER_OF_FARMERS_IN_VSLAS);
+            totalVslaFarmerMetricWrapper.setMetricId(String.valueOf(metricData.Id));
+            totalVslaFarmerMetricWrapper.setCurrentTarget(metricData.Projected_Value__c);
+            totalVslaFarmerMetricWrapper.setCurrentRealValue((Decimal) metricData.Manual_Value__c);
+            // TODO: Get these from db  
+            metricData = getMetricData(TOTAL_NUMBER_OF_MALE_FARMERS_IN_VSLAS);
+            totalVslaMaleFarmerMetricWrapper.setMetricId(String.valueOf(metricData.Id));
+            totalVslaMaleFarmerMetricWrapper.setCurrentTarget(metricData.Projected_Value__c);
+            totalVslaMaleFarmerMetricWrapper.setCurrentRealValue((Decimal) metricData.Manual_Value__c);
+            // TODO: Get these from db  
+            metricData = getMetricData(TOTAL_NUMBER_OF_FEMALE_FARMERS_IN_VSLAS);
+            totalVslaFemaleFarmerMetricWrapper.setMetricId(String.valueOf(metricData.Id));
+            totalVslaFemaleFarmerMetricWrapper.setCurrentTarget(metricData.Projected_Value__c);
+            totalVslaFemaleFarmerMetricWrapper.setCurrentRealValue((Decimal) metricData.Manual_Value__c);
+        } catch (NullPointerException ex) {
+            	System.debug('Error'+ ex.getMessage());
```

Comment:
log filters could skip this. it its always better you set the debugging level to ERROR